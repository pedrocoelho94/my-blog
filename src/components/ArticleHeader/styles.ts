import styled, { css } from 'styled-components'
import * as HeadingStyles from 'components/Heading/styles'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles.Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.font.sizes.large};
    }
  `}
`

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: calc(${theme.font.sizes.medium} * 0.9);
  `}
`

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`
