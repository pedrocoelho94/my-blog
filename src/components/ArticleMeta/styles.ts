import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child):after {
      content: ', ';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.2s ease-in-out;

      &:hover {
        filter: brightness(0.5);
      }
    }
  `}
`
