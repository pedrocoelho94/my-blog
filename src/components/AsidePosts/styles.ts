import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Content = styled.a`
  ${({ theme }) => css`
    display: flex;
    gap: 1rem;
    height: 100%;
    cursor: pointer;
    text-decoration: none;
    margin: ${theme.spacings.medium} 0;
  `}
`

export const Cover = styled.img`
  border-radius: 0.8rem;
  width: 10rem;
  height: 10rem;
  object-fit: cover;
`

export const Title = styled.h4`
  ${({ theme }) => css`
    font-size: 1.5rem;
    height: 100%;
    margin: 0;
    color: ${theme.colors.darker};

    &:hover {
      text-decoration: underline;
    }
  `}
`
