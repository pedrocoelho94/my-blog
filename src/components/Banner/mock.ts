import { data } from 'api/mockPosts.json'

export default {
  settings: data.setting,
  post: data.posts[0]
}
