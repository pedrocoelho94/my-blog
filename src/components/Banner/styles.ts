import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  position: relative;
`

export const Content = styled.div`
  filter: brightness(0.95);
  transition: filter 0.2s ease-in-out;

  &:hover {
    filter: brightness(1.2);
  }
`

type imgProps = {
  src: string
}

export const Image = styled.div<imgProps>`
  ${({ src }) => css`
    width: 100%;
    height: 30rem;
    background-image: url(${src});
    background-position: center center;
    background-size: cover;

    ${media.greaterThan('small')`
      height: 40rem;
    `}

    ${media.greaterThan('medium')`
      height: 45rem;
    `}
  `}
`
export const Caption = styled.div`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.medium};
    background-color: rgba(0, 0, 0, 0.4);
    position: absolute;
    bottom: 0;
    left: 0;
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    margin-top: 0;
    color: ${theme.colors.white};

    ${media.lessThan('small')`
      font-size: 2rem;
      margin-bottom: 0;
    `}
  `}
`

export const Subtitle = styled.h3`
  ${({ theme }) => css`
    font-weight: inherit;
    margin-bottom: 0;
    color: ${theme.colors.white};

    ${media.lessThan('small')`
      display: none
    `}
  `}
`
