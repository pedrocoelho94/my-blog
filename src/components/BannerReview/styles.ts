import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Poster = styled.div`
  position: relative;
  width: 100%;
  max-width: 25rem;
  height: 100%;
  max-height: 35rem;
  transition: all 0.2s ease-in-out;
  overflow: hidden;
  border-radius: 5px;
`

export const Image = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 5px;
  transition: all 0.2s ease-in-out;
  transform: scale(1.05);

  &:hover {
    transform: scale(1.1);
  }
`

type RatingProps = {
  rating: number
}

const RatingModifiers = {
  good: () => `
    background-color: #01d200;
  `,
  ok: () => `
    background-color: #ffb300;
  `,
  bad: () => `
    background-color: #ff7e00;
  `,
  terrible: () => `
    background-color: #df0700;
  `
}

export const Rating = styled.div<RatingProps>`
  ${({ theme, rating }) => css`
    position: absolute;
    bottom: 0;
    right: 0;
    z-index: 5;
    color: ${theme.colors.white};
    font-size: 1.8rem;
    font-weight: bold;
    width: 3.5rem;
    height: 3.5rem;
    display: flex;
    border-radius: 5px;

    span {
      width: 100%;
      text-align: center;
      justify-self: center;
      align-self: center;
    }

    ${rating >= 7.5 && RatingModifiers.good()}
    ${rating >= 5 && rating < 7.5 && RatingModifiers.ok()}
    ${rating >= 2.5 && rating < 5 && RatingModifiers.bad()}
    ${rating < 2.5 && RatingModifiers.terrible()}
  `}
`
