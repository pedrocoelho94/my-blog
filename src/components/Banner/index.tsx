import Link from 'next/link'
import { Category } from 'shared-typed/category'
import { StrapiImage } from 'shared-typed/cover'
import * as S from './styles'

export type BannerProps = {
  slug: string
  cover: StrapiImage
  title: string
  excerpt: string
  categories: Category[]
}

const Banner = ({ cover, slug, title, excerpt, categories }: BannerProps) => (
  <S.Wrapper>
    <Link href={`/${categories?.[0].slug}/${slug}`} passHref>
      <a>
        <S.Content>
          <S.Image src={cover.url} role="img" aria-label={title} />

          <S.Caption>
            <S.Title>{title}</S.Title>
            <S.Subtitle>{excerpt}</S.Subtitle>
          </S.Caption>
        </S.Content>
      </a>
    </Link>
  </S.Wrapper>
)

export default Banner
