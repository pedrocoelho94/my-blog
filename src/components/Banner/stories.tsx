import { Story, Meta } from '@storybook/react'
import Banner, { BannerProps } from '.'
import mock from './mock'

export default {
  title: 'Banner',
  component: Banner,
  args: {
    title: mock.post.title,
    excerpt: mock.post.excerpt,
    slug: mock.post.slug,
    cover: mock.post.cover
  }
} as Meta

export const Default: Story<BannerProps> = (args) => (
  <div style={{ maxWidth: '120rem', margin: '0 auto' }}>
    <Banner {...args} />
  </div>
)
