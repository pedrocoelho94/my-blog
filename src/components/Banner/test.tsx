import { screen } from '@testing-library/react'
import { renderWithTheme } from 'styles/renderWithTheme'

import Banner from '.'

const props = {
  title: 'Title',
  excerpt: 'Excerpt',
  slug: 'slug',
  cover: {
    id: '123123',
    alternativeText: 'Alternative text',
    url: 'image.jpg'
  },
  categories: [
    {
      id: 'sdfiosdjf',
      displayName: 'cat',
      slug: 'cat-slug'
    }
  ]
}

describe('<Banner />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Banner {...props} />)

    expect(screen.getByRole('heading', { name: /title/i })).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /excerpt/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /title/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
