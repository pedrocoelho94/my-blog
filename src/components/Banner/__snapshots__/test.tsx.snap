// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctly 1`] = `
.c0 {
  position: relative;
}

.c1 {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c1:hover {
  -webkit-filter: brightness(1);
  filter: brightness(1);
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
}

.c2 {
  width: 100%;
  height: 30rem;
  background-image: url(image.jpg);
  background-position: center center;
  background-size: cover;
}

.c3 {
  width: 100%;
  padding: 2.4rem;
  background-color: rgba(0,0,0,0.4);
  position: absolute;
  bottom: 0;
  left: 0;
}

.c4 {
  margin-top: 0;
  color: #FFFFFF;
}

.c5 {
  font-weight: inherit;
  margin-bottom: 0;
  color: #FFFFFF;
}

@media (min-width:450px) {
  .c2 {
    height: 40rem;
  }
}

@media (min-width:768px) {
  .c2 {
    height: 45rem;
  }
}

@media (max-width:450px) {
  .c4 {
    font-size: 2rem;
    margin-bottom: 0;
  }
}

@media (max-width:450px) {
  .c5 {
    display: none;
  }
}

<div
  class="c0"
>
  <a
    href="/undefined/slug"
  >
    <div
      class="c1"
    >
      <div
        aria-label="Title"
        class="c2"
        role="img"
        src="image.jpg"
      />
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Title
        </h2>
        <h3
          class="c5"
        >
          Excerpt
        </h3>
      </div>
    </div>
  </a>
</div>
`
